### DISCUSSION TITLE
Migrate Jenkins to GitHub actions

### DISCUSSION THREAD

**Discussion by bossenti** [apache/streampipes#842] [General]
I was just wondering if we can migrate our Jeninks job to GitHub actions?
As far as I can tell, this job deploys our `dev` branch as a snapshot version in Maven, right?
That should then be easy to map as a GitHub workflow as well, then we have everything in one place.
Are there any other CI or CD workflows that are not currently implemented via GitHub actions?

**Comment by dominikriemer** [comment]
The only reason we have the Jenkins build in place is that it wasn't allowed to publish artifacts to the ASF maven repo from Github as per Infra policy when we moved to the Incubator.
But it seems this has changed and we can get credentials to publish to Nexus:

https://issues.apache.org/jira/browse/INFRA-23810

So it seems we need to ask Infra to enable NEXUS_USER and NEXUS_PW and we can get rid of the Jenkins job :-)

**Comment by bossenti** [comment]
That sounds great!
Thanks a lot for researching.

In this case I would create an issue for the migration, so that we can tackle this after the current release.

**Reply by bossenti** [reply]
see #843
